src/Nethermind/Nethermind.Monitoring/Metrics/MetricsController.cs

```diff
@@ -66,7 +64,22 @@ private void CreateMemberInfoMectricsGauge(MemberInfo propertyInfo, string gauge
                     attribute => GetStaticMemberInfo(attribute.Informer, attribute.Label));
 
             string description = propertyInfo.GetCustomAttribute<DescriptionAttribute>()?.Description;
-            _gauges[gaugeName] = CreateGauge(BuildGaugeName(propertyInfo.Name), description, configuration);
+
+            MetricsManualNamedAttribute userNamed = propertyInfo
+                .GetCustomAttribute<MetricsManualNamedAttribute>();
+
+            Gauge gauge;
+            if (userNamed != null)
+            {
+                gauge = CreateGauge(userNamed.Name, description, configuration, true);
+            }
+            else
+            {
+                string defaultName = BuildGaugeName(propertyInfo.Name);
+                gauge = CreateGauge(defaultName, description, configuration);
+            }
+
+            return gauge;
```

Comment:
Along with previous comment all of this can be collapsed to:

```
return CreateGauge(BuildGaugeName(propertyInfo), description, configuration);

private string BuildGaugeName(propertyInfo) => propertyInfo.GetCustomAttribute<MetricsManualNamedAttribute>()?.Name ?? BuildGaugeName(propertyInfo.Name);
```

Reply:
Yes, it looks much better now. Thank you for the suggestion!